### DISCUSSION TITLE
Should I create new Instance of GraphQL class for each executionInput?

### DISCUSSION THREAD

**Discussion by shubham2047** [graphql-java/graphql-java#3649] [Q&A] [answered]
I can see in javadoc of GraphQL class  "Building this object is very cheap and can be done on each execution if necessary". Should I create new GraphQL class instance each time for executing new ExecutionInput instance? when it is necessary to create new instance?

**Comment by bbakerman** [comment, marked as answer]
The full quote is :

>  Building this object is very cheap and can be done on each execution if necessary.  Building the schema is often not
> as cheap, especially if it's parsed from graphql IDL schema format via {@link graphql.schema.idl.SchemaParser}.

So it depends on how your code based is structured

In a general sense you should NOT build a `graphql.GraphQL` object per request since the schema and intstrumentations etc... are likely to be the same per request.  It could be a singleton component say

However if there was a user case for different behaviors depending on say the user or path, then its cheap to create a `graphql.GraphQL` per request but typically not cheap to create a `graphql.schema.GraphQLSchema` per request

So as a general rule - use a singleton `graphql.GraphQL` object
